hydroflow_lang/src/graph/ops/buffer.rs

```diff
@@ -0,0 +1,123 @@
+    hard_range_out: RANGE_1,
+    soft_range_out: RANGE_1,
+    num_args: 1,
+    is_external_input: false,
+    ports_inn: None,
+    ports_out: None,
+    properties: FlowProperties {
+        deterministic: FlowPropertyVal::Preserve,
+        monotonic: FlowPropertyVal::No,
+        inconsistency_tainted: false,
+    },
+    input_delaytype_fn: |_| None,
+    write_fn: |wc @ &WriteContextArgs {
+                   ident,
+                   op_span,
+                   root,
+                   inputs,
+                   outputs,
+                   is_pull,
+                   op_inst: OperatorInstance { arguments, .. },
+                   ..
+               },
+               _| {
+        let internal_buffer = wc.make_ident("internal_buffer");
+
+        let receiver = &arguments[0];
+        let stream_ident = wc.make_ident("stream");
+
+        let write_prologue = quote_spanned! {op_span=>
+            let mut #stream_ident = ::std::boxed::Box::pin(#receiver);
+            let #internal_buffer = df.add_state(::std::cell::RefCell::new(::std::vec::Vec::new()));
+        };
+
+        let write_iterator = if is_pull {
+            let input = &inputs[0];
+
+            quote_spanned! {op_span=>
+
+                {
+                    let mut vec = context.state_ref(#internal_buffer).borrow_mut();
+                    vec.extend(#input);
+                }
+
+                let #ident = match #root::futures::stream::Stream::poll_next(#stream_ident.as_mut(), &mut ::std::task::Context::from_waker(&context.waker())) {
+                    ::std::task::Poll::Ready(_) => {
+                        let mut vec = context.state_ref(#internal_buffer).borrow_mut();
+                        ::std::mem::take(&mut *vec)
+                    },
+                    ::std::task::Poll::Pending => {
+                        ::std::vec::Vec::new()
+                    },
+                }.into_iter();
+            }
+        } else {
+            let output = &outputs[0];
+
+            quote_spanned! {op_span=>
+
+                let #ident = #root::pusherator::for_each::ForEach::new(|x| {
+                    let mut vec = context.state_ref(#internal_buffer).borrow_mut();
+
+                    vec.push(x);
+                });
+
+                {
+                    let mut out = #output;
+                    for x in match #root::futures::stream::Stream::poll_next(#stream_ident.as_mut(), &mut ::std::task::Context::from_waker(&context.waker())) {
+                        ::std::task::Poll::Ready(_) => {
+                            let mut vec = context.state_ref(#internal_buffer).borrow_mut();
+                            ::std::mem::take(&mut *vec)
+                        },
+                        ::std::task::Poll::Pending => {
+                            ::std::vec::Vec::new()
+                        },
+                    }.into_iter() {
+                        <_ as #root::pusherator::Pusherator>::give(&mut out, x);
```

Comment:
actually if it is just _ then it's equivalent to #root::pusherator::Pusherator>::give(&mut out, x);

Reply:
Oh ok yeah